dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java

```diff
@@ -1076,559 +1042,611 @@ private String getEventSelectQuery( EventSearchParams params, MapSqlParameterSou
+                    .append( PSI_STATUS_EQ )
+                    .append( ":" + PSI_STATUS )
+                    .append( " " );
+            }
+        }
+
+        return stringBuilder.toString();
+    }
+
+    private String addLastUpdatedFilters( EventSearchParams params, MapSqlParameterSource mapSqlParameterSource,
+        SqlHelper hlp, boolean useDateAfterEndDate )
     {
         StringBuilder sqlBuilder = new StringBuilder();
 
         if ( params.hasLastUpdatedDuration() )
         {
-            sqlBuilder.append( hlp.whereAnd() ).append( "'" ).append( PSI_LASTUPDATED_GT )
-                .append( getLongGmtDateString( DateUtils.nowMinusDuration( params.getLastUpdatedDuration() ) ) )
-                .append( "' " );
+            mapSqlParameterSource.addValue( "lastUpdated", DateUtils.offSetDateTimeFrom(
+                DateUtils.nowMinusDuration( params.getLastUpdatedDuration() ) ), Types.TIMESTAMP_WITH_TIMEZONE );
+
+            sqlBuilder.append( hlp.whereAnd() )
+                .append( PSI_LASTUPDATED_GT )
+                .append( ":lastUpdated" )
+                .append( " " );
         }
         else
         {
             if ( params.hasLastUpdatedStartDate() )
             {
-                sqlBuilder.append( hlp.whereAnd() ).append( "'" ).append( PSI_LASTUPDATED_GT )
-                    .append( DateUtils.getLongDateString( params.getLastUpdatedStartDate() ) ).append( "' " );
+                mapSqlParameterSource.addValue( "lastUpdatedStart", params.getLastUpdatedStartDate(), Types.TIMESTAMP );
+
+                sqlBuilder.append( hlp.whereAnd() )
+                    .append( PSI_LASTUPDATED_GT )
+                    .append( ":lastUpdatedStart" )
+                    .append( " " );
             }
 
             if ( params.hasLastUpdatedEndDate() )
             {
                 if ( useDateAfterEndDate )
                 {
-                    Date dateAfterEndDate = addDays( params.getLastUpdatedEndDate(), 1 );
-                    sqlBuilder.append( hlp.whereAnd() ).append( " psi.lastupdated < '" )
-                        .append( DateUtils.getLongDateString( dateAfterEndDate ) ).append( "' " );
+                    mapSqlParameterSource.addValue( "lastUpdatedEnd", addDays( params.getLastUpdatedEndDate(), 1 ),
+                        Types.TIMESTAMP );
+
+                    sqlBuilder.append( hlp.whereAnd() )
+                        .append( " psi.lastupdated < " )
+                        .append( ":lastUpdatedEnd" )
+                        .append( "' " );
                 }
                 else
                 {
-                    sqlBuilder.append( hlp.whereAnd() ).append( " psi.lastupdated <= '" )
-                        .append( DateUtils.getLongDateString( params.getLastUpdatedEndDate() ) ).append( "' " );
+                    mapSqlParameterSource.addValue( "lastUpdatedEnd",
+                        DateUtils.getLongDateString( params.getLastUpdatedEndDate() ), Types.TIMESTAMP );
```

Comment:
Do we need getLongDateString here? I see you removed it from the other places and used Types.TIMESTAMP

Reply:
ok thanks for spotting that was going to fail with merged tests